src/main/scala/com/github/traviscrawford/spark/dynamodb/DynamoDBRelation.scala

```diff
@@ -98,18 +99,6 @@ private[dynamodb] case class DynamoDBRelation(
       .flatMap(scan)
   }
 
-  override def getScanSpec(config: ScanConfig): ScanSpec = {
-    config.maybeRequiredColumns match {
-      case Some(requiredColumns) =>
-        val expressionSpecBuilder =
-          new ExpressionSpecBuilder().addProjections(requiredColumns: _*)
-        super
-          .getScanSpec(config)
-          .withExpressionSpec(expressionSpecBuilder.buildForScan())
-      case None => super.getScanSpec(config)
-    }
-  }
-
```

Comment:
Consolidated `ScanSpec` creation into the base class. I think this is OK to do because most of the configuration is wrapped in `Option`s